typescript/sdk/src/token/deploy.ts

```diff
@@ -140,54 +136,56 @@ export class HypERC20Deployer extends GasRouterDeployer<
     config: HypERC20CollateralConfig,
   ): Promise<HypERC20Collateral> {
     const isFast = isFastConfig(config);
-    const factory = isFast
-      ? new FastHypERC20Collateral__factory()
-      : new HypERC20Collateral__factory();
-    const name = isFast ? 'FastHypERC20Collateral' : 'HypERC20Collateral';
-
-    return this.deployContractFromFactory(chain, factory, name, [
-      config.token,
-      config.mailbox,
-    ]);
+    if (isFast) {
+      return this.deployContract<TokenType.fastCollateral>(
+        chain,
+        TokenType.fastCollateral,
+        [config.token, config.mailbox],
+      );
+    } else {
+      return this.deployContract<TokenType.collateral>(
+        chain,
+        TokenType.collateral,
+        [config.token, config.mailbox],
+      );
+    }
   }
 
   protected async deployNative(
     chain: ChainName,
     config: HypNativeConfig,
   ): Promise<HypNative> {
-    let router: HypNative;
     if (config.scale) {
-      router = await this.deployContractFromFactory(
+      return this.deployContract<TokenType.nativeScaled>(
         chain,
-        new HypNativeScaled__factory(),
-        'HypNativeScaled',
+        TokenType.nativeScaled,
         [config.scale, config.mailbox],
       );
     } else {
-      router = await this.deployContractFromFactory(
-        chain,
-        new HypNative__factory(),
-        'HypNative',
-        [config.mailbox],
-      );
+      return this.deployContract<TokenType.native>(chain, TokenType.native, [
+        config.mailbox,
+      ]);
     }
-    return router;
   }
 
   protected async deploySynthetic(
     chain: ChainName,
     config: HypERC20Config,
   ): Promise<HypERC20> {
-    const isFast = isFastConfig(config);
-    const factory = isFast
-      ? new FastHypERC20__factory()
-      : new HypERC20__factory();
-    const name = isFast ? 'FastHypERC20' : 'HypERC20';
-
-    const router = await this.deployContractFromFactory(chain, factory, name, [
-      config.decimals,
-      config.mailbox,
-    ]);
+    let router: HypERC20;
+    if (isFastConfig(config)) {
+      router = await this.deployContract<TokenType.fastSynthetic>(
+        chain,
+        TokenType.fastSynthetic,
+        [config.decimals, config.mailbox],
+      );
+    } else {
+      router = await this.deployContract<TokenType.synthetic>(
+        chain,
+        TokenType.synthetic,
+        [config.decimals, config.mailbox],
+      );
```

Comment:
ditto for here with just using isFast for switching on the key to `deployContract`

Reply:
done